src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -759,6 +759,22 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this, Enumerator.KeyValuePair);
         }
 
+        /// <summary>
+        /// Ensures that the dictionary can hold up to 'capacity' entries without any further expansion of its backing storage
+        /// </summary>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            if (_entries != null && _entries.Length >= capacity)
+                return _entries.Length;
+            if (_buckets == null)
+                Initialize(0);
+            int newSize = HashHelpers.ExpandPrime(capacity);
```

Comment:
This likely wants to be `GetPrime` rather than `ExpandPrime`? So it can return itself if its already (one of the limited) primes or nearest if not.

`ExpandPrime` will double the value first; then look for the prime.

Reply:
You might want to have a test for this actually, eg., EnsureCapacity(15) should return 17 and not 37

Reply:
And EnsureCapacity(17) should return 17 and not 37